ReSharper.FSharp/src/FSharp/FSharp.Psi.Services/src/Util/PsiUtil.fs

```diff
@@ -409,6 +409,19 @@ let rec skipIntermediateParentsOfSameType<'T when 'T :> ITreeNode and 'T: not st
 let rec skipIntermediatePatParents (fsPattern: IFSharpPattern) =
     skipIntermediateParentsOfSameType<IFSharpPattern> fsPattern
 
+let rec isPartOfDotLambda (expr: IFSharpExpression) =
```

Comment:
Wouldn't something like this work better?
```fsharp
let appExpr = PrefixAppExprNavigator.GetByArgumentExpression(expr)
let expr = if isNotNull appExpr then appExpr else expr
isNotNull (DotLambdaExprNavigator.BetByExpression(expr))
```

Reply:
I think this won't work for expressions like

`_.A.B<caret>.C.D(5)`, because parent for B is `A.B.C`.

Reply:
Then maybe we should have a helper to get a qualified expr by its qualifiers.